specs/HubProtocol.md

```diff
@@ -348,8 +363,160 @@ World;0:;
 
 Note that the final frame still ends with the `;` terminator, and that since the body may contain `;`, newlines, etc., the length is specified in order to know exactly where the body ends.
 
+## MessagePack (MsgPack) encoding
+
+In the MsgPack Encoding of the SignalR Protocol, each Message is represented as a single MsgPack array containing items that correspond to properties of the given hub protocol message. The array items may be primitive values, arrays (e.g. method arguments) or objects (e.g. argument value). The first item in the array is the message type.
```

Comment:
The interesting thing is that we could do the exact same thing with JSON for the exact same reasons, so why not do that too? Why not use arrays instead of objects with field names.